File: TheaterSuggestions/CSharp/SeatsSuggestions/Row.cs
namespace SeatsSuggestions;

public class Row(string name, List<Seat> seats)
{
    public string Name { get; } = name;
    public List<Seat> Seats { get; } = seats;

    public SeatingOptionSuggested SuggestSeatingOption(int partyRequested, PricingCategory pricingCategory)
    { 
        var seatAllocation = new SeatingOptionSuggested(partyRequested, pricingCategory);
        
        foreach (var seat in Seats)
        {
            if (seat.IsAvailable() && seat.MatchCategory(pricingCategory))
            {
                seatAllocation.AddSeat(seat);
                if (seatAllocation.MatchExpectation()) return seatAllocation;
            }
        }
        return new SeatingOptionNotAvailable(partyRequested, pricingCategory);
    }
}

File: TheaterSuggestions/Java/SeatingPlaceSuggestions/src/main/java/org/weaveit/seatingplacesuggestions/Row.java
package org.weaveit.seatingplacesuggestions;

import java.util.List;

public class Row {
    private String name;
    private final List<SeatingPlace> seatingPlaces;

    public Row(String name, List<SeatingPlace> seatingPlaces) {
        this.name = name;
        this.seatingPlaces = seatingPlaces;
    }

    public SeatingOptionSuggested suggestSeatingOption(int partyRequested, PricingCategory pricingCategory) {
        var seatAllocation = new SeatingOptionSuggested(partyRequested, pricingCategory);

        for (var seat : seatingPlaces) {
            if (seat.isAvailable() && seat.matchCategory(pricingCategory)) {
                seatAllocation.addSeat(seat);

                if (seatAllocation.matchExpectation())
                    return seatAllocation;

            }
        }
        return new SeatingOptionNotAvailable(partyRequested, pricingCategory);
    }

}
